USER_GUIDE.md

```diff
@@ -1,7 +1,166 @@
+
+##### Run
+The `run` configuration has the following parameters. Note that only a subset are valid for Dev App Server version "1"
+and all are valid for Dev App Server version "2-alpha"
+
+Valid for versions "1" and "2-alpha"
+
+| Parameter             | Description |
+| --------------------- | ----------- |
+| ~~`appYamls`~~        | Deprecated in favor of `services` |
+| `jvmFlags`            | JVM flags to pass to the App Server Java process. |
+| `host`                | Application host address. |
+| `port`                | Application host port. |
+| `startSuccessTimeout` | Amount of time in seconds to wait for the Dev App Server to start in the background. |
+| `serverVersion`       | Server versions to use, options are "1" or "2-alpha" |
+| `services`            | List of services to run |
+ 
+Only valid for versions "2-alpha"
+
+| Parameter (2-alpha only) |
+| ------------------------ |
+| `adminHost`              |
+| `adminPort`              |
+| `allowSkippedFiles`      |
+| `apiPort`                |
+| `authDomain`             |
+| `automaticRestart`       |
+| `clearDatastore`         |
+| `customEntrypoint`       |
+| `datastorePath`          |
+| `defaultGcsBucketName`   |
+| `devAppserverLogLevel`   |
+| `logLevel`               |
+| `maxModuleInstances`     |
+| `pythonStartupArgs`      |
+| `pythonStartupScript`    |
+| `runtime`                |
+| `skipSdkUpdateCheck`     |
+| `storagePath`            |
+| `threadsafeOverride`     |
+| `useMtimeFileWatcher`    |
+
+##### Stage
+The `stage` configuration has the following parameters.
+
+| Parameter               | Description |
+| ----------------------- | ----------- |
+| `compileEncoding`       | The character encoding to use when compiling JSPs. |
+| `deleteJsps`            | Delete the JSP source files after compilation. |
+| `disableJarJsps`        | Disable adding the classes generated from JSPs. |
+| `disableUpdateCheck`    | Disable checking for App Engine SDK updates. |
+| `enableJarClasses`      | Jar the WEB-INF/classes content. |
+| `enableJarSplitting`    | Split JAR files larger than 10 MB into smaller fragments. |
+| `enableQuickstart`      | Use Jetty quickstart to process servlet annotations. |
+| `jarSplittingExcludes`  | Exclude files that match the list of comma separated SUFFIXES from all JAR files. |
+| `sourceDirectory`       | The location of the compiled web application files, or the exploded WAR. This is used as the source for staging. |
+| `stagingDirectory`      | The directory to which to stage the application. |
+
+##### Deploy
+The `deploy` configuration has the following parameters, deploy has some Flexible environment only parameters that
+are not listed here and will just be ignored.
+
+| Parameter             | Description |
+| --------------------- | ----------- |
+| `appEngineDirectory`  | Location of configuration files (cron.yaml, dos.yaml, etc) for configuration specific deployments. |
```

Comment:
Isn't `appEngineDirectory` flex only? Also it's a parameter for staging in the maven plugin